### PR TITLE
bakery/mgostorage: do not copy sessions

### PR DESCRIPTION
**Pull request by rogpeppe**
This means we can create a new mgostorage without incurring
a database round trip.

### CODE REVIEW

**Comment by mhilton** [comment]
:+1: LGTM

File: bakery/mgostorage/storage.go [hunk 1]
```diff
@@ -6,21 +6,22 @@ package mgostorage
...
+// the mgo session associated with the collection.
 func New(c *mgo.Collection) (bakery.Storage, error) {
 	m := mgoStorage{
 		col: c,
 	}
-	err := m.setUpCollection()
-	if err != nil {
-		return nil, errgo.Mask(err)
-	}
 	return &m, nil
```
Comment (alesstimec) [inline bakery/mgostorage/storage.go]: return &mgoStorage{col:c}, nil

File: bakery/mgostorage/storage.go [hunk 2]
```diff
@@ -29,43 +30,25 @@ type mgoStorage struct {
 }
 
 type storageDoc struct {
-	Location string `bson:"loc"`
+	Location string `bson:"_id"`
 	Item     string `bson:"item"`
-}
 
-func (s *mgoStorage) setUpCollection() error {
```
Comment (alesstimec) [inline bakery/mgostorage/storage.go]: Does this mean the caller is now responsible for setting up the collection index? I'm not completely sure i like this..

**Comment by alesstimec** [comment]
LGTM with a nitpick

File: bakery/mgostorage/storage.go (hunk 1 above)
Comment (rogpeppe) [inline bakery/mgostorage/storage.go]: done, slightly differently (no need to allocate a pointer)

**Pull request merged**